assert/assertions.go

```diff
@@ -806,10 +806,11 @@ func NotContains(t TestingT, s, contains interface{}, msgAndArgs ...interface{})
 
 }
 
-// Subset asserts that the specified list(array, slice...) contains all
-// elements given in the specified subset(array, slice...).
+// Subset asserts that the specified list(array, slice...) or map contains all
+// elements given in the specified subset list(array, slice...) or map.
 //
 //    assert.Subset(t, [1, 2, 3], [1, 2], "But [1, 2, 3] does contain [1, 2]")
+//    assert.Subset(t, {"x": 1, "y": 2}, {"x": 1}, "But {"x": 1, "y": 2} does contain {"x": 1}")
```

Comment:
The code in `assertion_forward.go` and `assertion_format.go` is auto-generated :eyes: 

Not quite sure why you pointed out the issues in those files, since `assertions.go` is the only source of truth here.